Content/Items/Infernal/InfernalHarvest.cs

```diff
@@ -0,0 +1,563 @@
+
+		private List<Vector2> cacheBig;
+		private Trail trailBig;
+
+		public Player Owner => Main.player[Projectile.owner];
+
+		public ref float Timer => ref Projectile.ai[0];
+		public ref float InvertDirection => ref Projectile.ai[1];
+
+		public override string Texture => "StarlightRiver/Assets/Items/Infernal/InfernalHarvestProj";
+
+		public override void SetDefaults()
+		{
+			Projectile.friendly = true;
+			Projectile.hostile = false;
+			Projectile.timeLeft = 110;
+			Projectile.width = SCYTHE_LENGTH * 2;
+			Projectile.height = SCYTHE_LENGTH * 2;
+			Projectile.tileCollide = false;
+			Projectile.penetrate = -1;
+			Projectile.extraUpdates = 1;
+			Projectile.DamageType = DamageClass.Magic;
+		}
+
+		public override void AI()
+		{
+			Timer++;
+
+			Projectile.Center = Owner.MountedCenter;
+
+			Owner.heldProj = Projectile.whoAmI;
+			Owner.SetCompositeArmFront(true, Player.CompositeArmStretchAmount.ThreeQuarters, Projectile.rotation - 1.57f);
+			Owner.SetCompositeArmBack(true, Player.CompositeArmStretchAmount.Full, Projectile.rotation - 1.97f);
+
+			if (Timer == 1)
+			{
+				direction = Projectile.velocity.X > 0 ? -1 : 1;
+				direction *= (int)InvertDirection;
+
+				startRot = Projectile.velocity.ToRotation() + 2f * direction;
+				Projectile.rotation = startRot;
+			}
+
+			if (Timer > 16)
+				Projectile.rotation = startRot - Ease((Timer - 16) / 94f) * 4 * direction;
+
+			if (Timer > 16 && Timer < 50)
+			{
+				Vector2 pos = Vector2.UnitX.RotatedBy(Projectile.rotation) * SCYTHE_LENGTH * 0.7f + Main.rand.NextVector2Circular(16, 16);
+				var d = Dust.NewDustPerfect(Projectile.Center, ModContent.DustType<Dusts.GlowFollowPlayer>(), Main.rand.NextVector2Circular(2, 2), 0, new Color(255, 150, 0) * 0.5f, Main.rand.NextFloat(0.3f, 0.6f));
+				d.customData = new object[] { Owner, pos };
+			}
+
+			if (Main.netMode != NetmodeID.Server)
+			{
+				ManageCaches();
+				ManageTrail();
+			}
+		}
+
+		public float Ease(float x)
+		{
+			return x == 1 ? 1 : 1f - (float)Math.Pow(2, -10 * x);
+		}
+
+		public override bool? CanHitNPC(NPC target)
+		{
+			Vector2 perp = Vector2.UnitX.RotatedBy(Projectile.rotation + 1.57f);
+
+			bool hit = target.immune[0] <= 0 && Timer < 60 && (
+				Helper.CheckLinearCollision(Projectile.Center, Projectile.Center + Vector2.UnitX.RotatedBy(Projectile.rotation) * SCYTHE_LENGTH, target.Hitbox, out _) ||
+				Helper.CheckLinearCollision(Projectile.Center + perp * 10, Projectile.Center + Vector2.UnitX.RotatedBy(Projectile.rotation) * SCYTHE_LENGTH + perp * 10, target.Hitbox, out _) ||
+				Helper.CheckLinearCollision(Projectile.Center + perp * -10, Projectile.Center + Vector2.UnitX.RotatedBy(Projectile.rotation) * SCYTHE_LENGTH + perp * -10, target.Hitbox, out _));
+
+			return hit ? null : false;
+		}
+
+		public override void OnHitNPC(NPC target, NPC.HitInfo hit, int damageDone)
+		{
+			int amount = Math.Min(20, Owner.statManaMax2 - Owner.statMana);
+
+			if (amount > 0)
+			{
+				Owner.statMana += amount;
```

Comment:
would desync mana in multiplayer. probably not critical but this should still send an onhit packet (and you get the sounds that way)

i.e. add 			
                        Owner.TryGetModPlayer(out StarlightPlayer starlightPlayer);
			starlightPlayer.SetHitPacketStatus(shouldRunProjMethods: true);

like echochain onhit as an example